.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,53 @@
+# Copyright The FMS HF Tuning Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build and Publish FMS-hf-tuning Library
+
+on:
+    workflow_dispatch:
+        inputs:
+            release-version:
+              type: string
+              required: true
+              description: 'Version number (for example: 0.0.1)'
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    environment:
+      name: pypi
+      url: https://pypi.org/p/fms-hf-tuning
+    permissions:
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r setup_requirements.txt
+      - name: Replace release-version with provided input value
+        run: |
+          export RELEASE_VERSION=${{ github.event.inputs.release-version }}
+          sed -i "s/^version = \"[0-9.]*\"$/version = \"${RELEASE_VERSION}\"/g" pyproject.toml
```

Comment:
Perhaps we should commit this change to the file or conversely, do it in the way, it's done the [here](https://github.com/opendatahub-io/caikit-nlp-client/blob/main/pyproject.toml).

Reply:
i could not find `version =`  is matched in [pyproject.toml](https://github.com/opendatahub-io/caikit-nlp-client/blob/main/pyproject.toml)

Reply:
you may find `version = ` in this [pyproject.toml](https://github.com/foundation-model-stack/fms-hf-tuning/blob/445c17b7c0bc6bd87794334d2eab71b495382f2b/pyproject.toml#L8)

Reply:
OK - working on this... thanks for the advice, reviews!